src/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,12 +503,37 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
             });
         }
 
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actionExitCodeId">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job,
+            Dictionary<ValueTuple<string, List<int>>, ValueType> actions)
+        {
+            job.Spec ??= new V1JobSpec();
+            job.Spec.PodFailurePolicy ??= new V1PodFailurePolicy();
+
+            job.Spec.PodFailurePolicy.Rules = actions.Select(action => new V1PodFailurePolicyRule
+            {
+                Action = action.Key.Item1,
+                OnExitCodes = new V1PodFailurePolicyOnExitCodesRequirement
+                {
+                    Values = action.Key.Item2.Distinct().ToList()
+                }
+            }).ToList();
+
+            return job;
+        }
+
         /// <summary>
         /// Clones a job object.
         /// </summary>
         /// <param name="job">V1Job to clone.</param>
         /// <returns>New V1Job object.</returns>
-        public static V1Job Clone(this V1Job job) => JsonSerializer.Deserialize<V1Job>(JsonSerializer.Serialize(job));
+        public static V1Job Clone(this V1Job job) =>
```

Comment:
funny enough, we have `DeepClone` method that does the same - maybe create an issue to remove this one in a separate PR

Reply:
I've created the issue https://github.com/SneaksAndData/esd-services-sdk/issues/77 and I'll take it in a separate PR after merging this one 👍